### PR TITLE
Framework: Remove `enzyme` from Calypso client

### PR DESCRIPTION
**Pull request by tyxla**
#### Proposed Changes

This PR removes `enzyme` from the Calypso client and Calypso's documentation, since it's now only used in a few apps and packages.

#### Testing Instructions

Verify all checks are green.

Related: #63409

cc @flootr

### CODE REVIEW

**Comment by tyxla** [comment]
Ready for review!

File: docs/testing/component-tests.md [hunk 1]
```diff
@@ -21,14 +21,14 @@ This obviously varies between components, but a few easy things to start out wit
 Like its child components?
 
 ```javascript
-import AnExpectedChildComponent from 'an-expected-child-component';
-import { shallow } from 'enzyme';
+import Dialog from '@automattic/components';
```
Comment (noahtallen) [inline docs/testing/component-tests.md]: Would importing `Dialog` be important for this test, assuming `MyComponent` already imports it and we just look it up by role?

**Review by noahtallen — APPROVED** [review]
With the tests passing, this looks good to go!

File: docs/testing/component-tests.md (hunk 1 above)
Comment (tyxla) [reply docs/testing/component-tests.md]: Good eye, I think it can be removed 👍

Comment (tyxla) [reply docs/testing/component-tests.md]: Done in 76d0a2529287055d3a5e0f5082d666adabb48339 ✅

**Pull request merged**